app/src/features/apiClient/screens/apiClient/components/modals/exportCollectionsModal/ExportCollectionsModal.tsx

```diff
@@ -0,0 +1,154 @@
+  records: (RQAPI.CollectionRecord | RQAPI.ApiRecord)[];
+}
+
+export const ExportCollectionsModal: React.FC<ExportCollectionsModalProps> = ({ isOpen, onClose, collections }) => {
+  const [isApiRecordsProcessed, setIsApiRecordsProcessed] = useState(false);
+  const [exportData, setExportData] = useState<ExportData>({ variables: [], records: [] });
+  const [isExportVariablesChecked, setIsExportVariablesChecked] = useState(true);
+  const { getVariableData } = useEnvironmentManager();
+
+  const handleExport = useCallback(() => {
+    const dataToExport = exportData;
+    if (!isExportVariablesChecked) {
+      dataToExport.variables = [];
+    }
+    trackExportApiCollectionsStarted(dataToExport.records.length);
+    try {
+      const fileContent = JSON.stringify(dataToExport, null, 2);
+      const fileName = `RQ-${collections.length === 1 ? "collection" : "collections"}-export-${getFormattedDate(
+        "DD_MM_YYYY"
+      )}.json`;
+      fileDownload(fileContent, fileName, "application/json");
+      onClose();
+      trackExportApiCollectionsSuccessful(dataToExport.records.length);
+    } catch (error) {
+      trackExportApiCollectionsFailed(dataToExport.records.length);
+    }
+  }, [exportData, isExportVariablesChecked, collections, onClose]);
+
+  const extractVariablesFromAPIRecord = useCallback(
+    (api: RQAPI.ApiRecord) => {
+      const variables = new Set<string>();
+      const apiRequest = api.data.request;
+
+      const checkAndAddVariable = (value: string) => {
+        const variable = extractVariableNameFromStringIfExists(value);
+        if (variable) variables.add(variable);
+      };
+
+      apiRequest.headers?.forEach(({ key, value }) => {
+        checkAndAddVariable(key);
+        checkAndAddVariable(value);
+      });
+
+      apiRequest.queryParams?.forEach(({ key, value }) => {
+        checkAndAddVariable(key);
+        checkAndAddVariable(value);
+      });
+
+      checkAndAddVariable(apiRequest.url);
+
+      if (isArray(apiRequest.body)) {
+        apiRequest.body.forEach((body) => {
+          checkAndAddVariable(body.value);
+        });
+      } else {
+        checkAndAddVariable(apiRequest.body || "");
+      }
+
+      return Array.from(variables).map(getVariableData);
+    },
+    [getVariableData]
+  );
+
+  useEffect(() => {
+    if (isOpen && !isApiRecordsProcessed) {
+      const extractedVariables: VariableExport[] = [];
+      const processedApiRecords: (RQAPI.CollectionRecord | RQAPI.ApiRecord)[] = [];
+
+      collections.forEach((collection) => {
+        const collectionToExport = { ...collection, data: {} };
+        processedApiRecords.push(collectionToExport);
+
+        collection.data.children.forEach((api) => {
+          processedApiRecords.push(api);
+          extractedVariables.push(...extractVariablesFromAPIRecord(api as RQAPI.ApiRecord));
+        });
+      });
+
+      setExportData({ variables: extractedVariables, records: processedApiRecords });
```

Comment:
Sanitize the data before exporting, refer [this](https://github.com/requestly/requestly/pull/2290#pullrequestreview-2437322032).